content/posts/2019-11-07-ln-split-tickets-decred-blog.md

```diff
@@ -0,0 +1,159 @@
+
+`SIGHASH_NOTOUTPUTVALUE` effectively makes the wallet sign the transaction discarding the output value. But consensus rules on the implementation of `OP_CHECKOUTPUTPERCENTAGE` would ensure the output is _at least_ the specified percentage of the corresponding input amount guaranteeing the users are getting back their relative contribution amounts plus their reward in case of votes.
+
+There are _many_ nuances to implementing these features and how they are effectively applied to allow the MRTTREE construction to be used in multi-owner tickets. There are also plenty of additional considerations for on-chain voting rights, Politeia voting rights and privacy improvements so for deeper look please see my [second post on LN and Split Ticket](https://matheusd.com/post/ln-split-tickets-02-splits).
+
+# Improving Stake Transactions
+
+The current consensus-enforced layout of ticket transactions carries quite some baggage and is much less efficient than it could be. The present ecosystem of Decred software makes the majority of its functions simply obsolete while still lacking features that would improve the blockchain load and allow better support of split tickets.
+
+Therefore assuming the multi-owner tickets will be moving to the LN version in the near future and will see increased usage we also propose a set of changes to improve to story of ticket transactions in Decred.
+
+The main change is to do away with the requirement that the number of outputs must be equal to twice the number of inputs (plus the voting output).
+
+For the majority of ticket purchases the corresponding `SSTX_CHANGE` outputs are always empty ([example](https://dcrdata.decred.org/tx/a8efdfb5f475f780a33289c847070a743de809d7db4d4b84e2c15dea8121b7c8/out/2)). And given the most recent privacy work also relying on split transactions for ticket purchases (that is a transaction which creates outputs of the exact required amount for a ticket purchase - [example](https://dcrdata.decred.org/tx/b2997108c53b78b5ac5bdfd7443e592d94d2a5dda706a87544749eafa1ad3e9e/out/0)) it's unlikely that type of output will be used in the future.
+
+Besides that and even without LN multi-owner tickets, being able to create fan-in and fan-out ticket purchases (that is, ticket purchases that consolidate funds or spread them over multiple redeeming UTXOs) would enable novel methods to build cooperative tickets.
+
+Another aspect that needs to be addressed is the need to improve handling of Politeia voting rights. Today those rights are assigned to the largest (by value) commitment of a ticket purchase, which makes them vulnerable to influence amplification. Having a separate output for an individual key would allow it to be assigned in a pseudo-random fashion or even with fractional voting power in case of a group key.
```

Comment:
```suggestion
Another aspect that needs to be addressed is the need to improve handling of Politeia voting rights. Today those rights are assigned to the largest (by value) commitment of a ticket purchase, which makes them vulnerable to influence amplification. Having a separate output for an individual key would allow it to be assigned in a pseudo-random fashion, or even with fractional voting power in case of a group key.
```